DigitalExcellence/dex-frontend
# General QoL fixes and fixes for Project Edit page

## Description

This pull requests fixes a few issues I've noticed with the current platform:

- "Add project" button on home page now works
- "Edit project" typescript / HTML was changed and should not function properly
- Buttons on the detailed project page like "edit project" and "highlight project" have received at least some styling. This was done so they don't look like the general basic HTML buttons.
- Buttons on the detailed project page are only shown for authenticated users.
- Project can be highlighted (user does not receive feedback at this moment), highlights are also permanent. When #108 is implemented, it can be done properly.

## Type of change


## Checklist


## Steps to Test or Reproduce

Project Details page Buttons

1. Go to project detail page without logging in
2. See if there are buttons, non should show up.
3. Login using "bob" username and "bob" password
4. Go to project detail page while bein logged in
5. Edit project and Highlight project button should show up.

Project Edit page

1. Go to project detail page while being logged in
2. Click the edit project button to edit the project
3. Add collaborators, remove collaborators and make changes to fields
4. Save your changes. You will notice it returns a 500 internal server error, this is because of an issue in the backend.


## Link to issue

Closes: No issue number since these were hotfixes for demo
